pkg/scalers/cpu_memory_scaler.go

```diff
@@ -59,10 +59,12 @@ func parseResourceMetadata(config *ScalerConfig) (*cpuMemoryMetadata, error) {
 		if err != nil {
 			return nil, err
 		}
-		utilizationNum, ok := valueNum.(int32)
+		// ParseNumeric returns int64, ideally, but we need int32.  Confirm it's an int64 first, then cast to int32.
+		_, ok := valueNum.(int64)
 		if !ok {
-			return nil, fmt.Errorf("provided value for Utilization (%d) was not a valid 32-bit integer", valueNum)
+			return nil, fmt.Errorf("provided value for Utilization (%d) was not a valid integer", valueNum)
 		}
+		utilizationNum := int32(valueNum.(int64))
```

Comment:
how about converting to int64 above, then you don't have to do the `valueNum.(int64)` again here?

```go
valueNum64, ok := valueNum.(int64)
if !ok { 
    //...
}
utilizationNum := int32(valueNum64)
```

Also, do we need to worry about overflow here?

Reply:
I thought about that and tried it and the issue is that there's the possibility that the user could pass a numeric that's a float, and `ParseNumeric()` will return a `float64`.  The way you wrote the code above is how I had written it previously and it actually broke the tests.  So the initial assertion on 63 is to make sure that we actually got an `int64` back and can return the error if it wasn't one.  We don't want to coerce something that the user says is a float into an int.  Then, if it did assert as an `int64`, we need to convert it to `int32` for it to actually be usable, which necessitates a type assertion because `ParseNumeric()` returns an `interface{}`.  

We don't have to worry about overflow (and for the same reason as the original code).  On 58, the last arg to `ParseNumeric()` is the bit length of the value to fit the result into.  If you pass a number that's larger than the maximum size it'd allow for, `ParseInt()` inherently changes it to be the maximum integer that the data type would allow for.  Under the hood, `ParseNumeric()` is just calling `ParseInt()` after validating that the passed value is an integer, so we inherit its behavior.

Reply:
Oh, apologies, I'm tracking what you're saying now.  Sorry, memory's fuzzy from originally doing this like a month ago.  Yeah, I think I can do that change and have it work.

Reply:
Though, actually, is that worth it?  Is it preferred to allocate more memory for the new variable instead of just doing the runtime assertion a second time?

Reply:
up to you. I was suggesting it for a bit of readability improvement, which is subjective. performance-wise, in the grand scheme of things, it's a very minimal amount of memory. without profiling, I couldn't really say if it has an impact at all.